library/Garp/Functional/Divide.php

```diff
@@ -0,0 +1,21 @@
+<?php
+/**
+ * @package  Garp\Functional
+ * @author   Harmen Janssen <harmen@grrr.nl>
+ * @license  https://github.com/grrr-amsterdam/garp-functional/blob/master/LICENSE.md BSD-3-Clause
+ */
+namespace Garp\Functional;
+
+/**
+ * Divide two numbers.
+ *
+ * @param int $left
+ * @param int $right
+ * @return int
+ */
+function divide($left, $right = null) {
+    $divider = function ($right) use ($left) {
+        return $left / $right;
```

Comment:
I think the way you're dividing here is not taking full advantage of currying.  
Because you're doing `$left / $right`, the order of arguments is no longer data-last. To me it makes more sense to do `$right / $left` (see also my `subtract` function), because it allows you to create building blocks, thanks to currying:

```php
$half = divide(2);
$divideBy10 = divide(10);

$half(40); // 20
$half(5); // 2.5
$divideBy10(100); // 10
```

Would you agree?